src/components/layout/Widget.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react'
+import styled from 'styled-components'
+import { depositApi } from 'api'
+import { EtherscanLink } from 'components/EtherscanLink'
+
+const Wrapper = styled.section`
+  font-size: 0.85rem;
+  padding-bottom: 4em;
+  padding: 2em;
+  border-radius: 10px;
+  min-width: 40vw;
+  background-color: white;
+  margin: -3rem auto 3rem auto;
+  box-shadow: 1px 1px #e8e8e8;
+  min-height: 25rem;
+
+  display: flex;
+  flex-direction: column;
+`
+
+const LinkWrapper = styled(EtherscanLink)`
+  text-align: right;
+  margin-bottom: 3em;
+  display: block;
+`
+
+const Widget: React.FC = ({ children }) => {
+  const contractAddress = depositApi.getContractAddress()
+
+  const contractLink = (
+    <LinkWrapper type="contract" identifier={contractAddress} label={<small>View verified contract</small>} />
+  )
+
+  return (
+    <Wrapper>
+      {contractLink && contractLink}
```

Comment:
`contractLink` is always true

Reply:
Depending on the network, `EtherscanLink` component might be null: https://github.com/gnosis/dex-react/blob/develop/src/components/EtherscanLink.tsx#L31-L36

Reply:
No, `EtherscanLink` may render to null, not be null. Component is always an object
```js
<EtherscanLink/> = React.createElement(EtherscanLink) = {type: EtherscanLink, ref, key, props: {}, ...metaStuff}

```

Reply:
Ah, so this is effectively the same as rendering the component inside the `Wrapper`. When `Etherscan` renders to `null` it simply won't show up:

```
    <Wrapper>
      <LinkWrapper type="contract" identifier={contractAddress} label={<small>View verified contract</small>} />
      {children}
    </Wrapper>
```